ggml/src/ggml-sycl/gemm.hpp

```diff
@@ -23,6 +23,10 @@
 #include "dnnl.hpp"
 #include "dnnl_sycl.hpp"
 
+
+// TODO: Remove this when needed
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wcast-qual"
```

Comment:
I think it's best to use `const_cast` when needed. From what I understand the const cast is needed as it is not possible to create constant oneDNN memory objects but we make sure to only read from these memory object. This reduces the risk of hiding other warnings.